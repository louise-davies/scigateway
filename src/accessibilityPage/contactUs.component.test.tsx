import React from 'react';
import { shallow } from 'enzyme';
import { UnconnectedContactUs, ContactUsProps } from './contactUs.component';

describe('Contact us component', () => {
  let props: ContactUsProps;

  it('renders iframe correctly if form url set', () => {
    props = {
      contactUsAccessibilityFormUrl: 'test-url',
    };

    const wrapper = shallow(<UnconnectedContactUs {...props} />);

    expect(wrapper.find('#contact-us-form')).toBeTruthy();
  });

  it('renders mailto link correctly if form url not set', () => {
    props = {
      contactUsAccessibilityFormUrl: '',
    };
    let wrapper = shallow(<UnconnectedContactUs {...props} />);
    expect(wrapper.find('#contact-info')).toBeTruthy();

    props = {
      contactUsAccessibilityFormUrl: undefined,
    };
    wrapper = shallow(<UnconnectedContactUs {...props} />);
    expect(wrapper.find('#contact-info')).toBeTruthy();
  });
});
