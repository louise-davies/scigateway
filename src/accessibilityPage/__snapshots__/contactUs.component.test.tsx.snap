// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact us component renders iframe correctly if form url set 1`] = `
<div
  id="contact-us"
>
  <div
    id="contact-us-form"
  >
    <iframe
      allowFullScreen={true}
      height="840px"
      src="test-url"
      title="Contact Us"
      width="740px"
    />
  </div>
</div>
`;

exports[`Contact us component renders mailto link correctly if form url not set 1`] = `
<div
  id="contact-us"
>
  <div
    id="contact-info"
  >
    <WithStyles(ForwardRef(Typography))
      className="description-class"
    >
      <WithStyles(ForwardRef(Link))
        href="mailto:accessibility-page.contact-info"
      >
        accessibility-page.contact-info
      </WithStyles(ForwardRef(Link))>
    </WithStyles(ForwardRef(Typography))>
  </div>
</div>
`;
